### PR TITLE
Doc update for oneTBB 2022.0

### PR DESCRIPTION
**Pull request by aepanchi**
### Description 
* Update the Get Started Guide to improve users experience
* Remove any mentions of 32-bit binaries
* Remove mention of captured_exception

### Type of change

_Choose one or multiple, leave empty if none of the other choices apply_

_Add a respective label(s) to PR if you have permissions_

- [ ] bug fix - _change that fixes an issue_
- [ ] new feature - _change that adds functionality_
- [ ] tests - _change in tests_
- [ ] infrastructure - _change in infrastructure and CI_
- [x] documentation - _documentation update_

### Tests

- [ ] added - _required for new features and some bug fixes_
- [x] not needed

### Documentation

- [ ] updated in # - _add PR number_
- [ ] needs to be updated
- [x] not needed

### Breaks backward compatibility
- [ ] Yes
- [x] No
- [ ] Unknown

### Notify the following users
_List users with `@` to send notifications_

### CODE REVIEW

File: doc/GSG/get_started.rst [hunk 1]
```diff
@@ -8,11 +8,36 @@ It is helpful for new users of parallel programming and experienced developers t
 
 It is recommended for you to have a basic knowledge of C++ programming and some experience with parallel programming concepts. 
 
+|full_name| is a runtime-based parallel programming model for C++ code that uses threads.
```
Comment (vossmjp) [inline doc/GSG/get_started.rst:11]:
```suggestion
|full_name| is a runtime-based parallel programming model for C++ code that uses tasks.
```

File: doc/GSG/get_started.rst [hunk 2]
```diff
@@ -8,11 +8,36 @@ It is helpful for new users of parallel programming and experienced developers t
...
+The template-based runtime library can help you harness the latent performance of multi-core processors.
+
+oneTBB enables you to simplify parallel programming by breaking computation into parallel running tasks. Within a single process, 
+parallelism is carried out through threads, an operating system mechanism that allows the same or different sets of instructions
```
Comment (vossmjp) [inline doc/GSG/get_started.rst:15]:
```suggestion
parallelism is carried out by mapping tasks to threads. Threads are an operating system mechanism that allows the same or different sets of instructions 
```

**Review by vossmjp — COMMENTED** [review]
A couple of suggestions

**Review by omalyshe — APPROVED** [review]
Looks good.

**Pull request merged**